### ISSUE TITLE
Support for xonsh

### ISSUE THREAD

**Issue by boborbt** [warpdotdev/Warp#2735]
### Describe the solution you'd like?

```text
Currently there is no support for xonsh, which is understandable. The main problem, though, is that warp defaults to zsh if the main shell of the user is set to xonsh. This is unreasonable IMHO, xonsh works perfectly fine in warp. Yes, it lacks many (all) distinguishing warp features, but if warp should respect the user preference anyway.

In synthesis: full support for xonsh would be great, but at least allowing the user to use it should be granted.
```

### Is your feature request related to a problem? Please describe.

```text
I'm always frustrated when I cannot used my favourite shell (even if this means a degraded experience out of warp).
```

### How important is this feature to you?

5 (Can't work without it!)

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
Thanks for submitting @boborbt!
We have an issue tracking this request here: 
https://github.com/warpdotdev/Warp/issues/880#issuecomment-1330917404

TLDR, we primarily support bash, zsh, and fish out of the box. Given enough user demand ( or releasing Warp for other platforms like Windows for example ), we'd certainly support other shells.

Please :+1: and subscribe to that request as it helps us gauge interest. We'll post any updates there.
Closing as duplicate.

**Comment by boborbt** [comment]
Hi Danny, thanks for the feedback.

Please notice that the main point in my request was a little different though, while I will be very happy to see support for other shells, the main point was another: please do not default to zsh (or any other shell) if the user happen to have set their default shell to something else. Respect your users’ preferences, maybe alert them of all the benefits they will be missing, but leave the ultimate choice of the shell to use to them. As always in programming nothing is as simple as it seems, but this really seems a 15 minutes update.

Hope this clarifies. Best,
Roberto
